### ISSUE TITLE
Unordered bulkwrite operations not persisting remaining writes in queue in event of exception 

### ISSUE THREAD

**Issue by tobiade** [bwaldvogel/mongo-java-server#142]
Hello,
According to the Mongo docs:

> unordered operations continue to process any remaining write operations in the queue, unless when run inside a transaction

I would expect the following piece of code to persist documents with IDs 1,2,4 as a duplicate key exception would be thrown for the duplicate document with ID 2.

However, it seems only the documents with ID 1 and 2 are persisted - the last document with ID 4 is not.

Any thoughts on why that is? If you could point me to which part of the code handles these bulk writes that'll be helpful too.
```
public class BulkWriteTest {

    private MongoCollection<Document> collection;
    private MongoClient client;
    private MongoServer server;

    @Before
    public void setUp() {
        server = new MongoServer(new MemoryBackend());
        InetSocketAddress serverAddress = server.bind();
        client = new MongoClient(new ServerAddress(serverAddress));
        collection = client.getDatabase("testdb").getCollection("testcollection");
    }

    @After
    public void tearDown() {
        client.close();
        server.shutdown();
    }

    @Test
    public void testBulWrite() {
        List<WriteModel<Document>> writes = new ArrayList<>();
        writes.add(new InsertOneModel<>(new Document("_id", 1)));
        writes.add(new InsertOneModel<>(new Document("_id", 2)));
        writes.add(new InsertOneModel<>(new Document("_id", 2)));
        writes.add(new InsertOneModel<>(new Document("_id", 4)));
        try {
            collection.bulkWrite(writes, new BulkWriteOptions().ordered(false));
        } catch (MongoBulkWriteException e) {
            e.printStackTrace();
            List<Document> docsPersisted = new ArrayList<>();
            collection.find().iterator().forEachRemaining(docsPersisted::add);
            assertEquals(3, docsPersisted.size());
        }
    }

}
```

**Comment by utkarsh-harness** [comment]
I am facing the same issue. This functionality used to work before and is broken now. The continueOnError flag in the InsertOptions is not being respected any more.

**Comment by bwaldvogel** [comment]
Thanks for reporting. It should be fixed in 237e96af754956f98987a8bad5e3829871e8325e. Can you re-test?

**Comment by tobiade** [comment]
Thanks @bwaldvogel. Left a [comment](https://github.com/bwaldvogel/mongo-java-server/commit/237e96af754956f98987a8bad5e3829871e8325e#r39754994), but aside from that, all looks good.

**Comment by tobiade** [comment]
Oops just seen your latest comment! Looks good - Thanks!

**Comment by bwaldvogel** [comment]
The fix is available in version `1.30.0`
